hudi-common/src/main/java/org/apache/hudi/common/table/TableSchemaResolver.java

```diff
@@ -232,7 +232,7 @@ private Option<Schema> getTableSchemaFromLatestCommitMetadata(boolean includeMet
       String schemaStr = commitMetadata.getMetadata(HoodieCommitMetadata.SCHEMA_KEY);
       Schema schema = new Schema.Parser().parse(schemaStr);
       if (includeMetadataFields) {
-        schema = HoodieAvroUtils.addMetadataFields(schema, hasOperationField.get());
+        schema = HoodieAvroUtils.addMetadataFields(schema, schema.getField(HoodieRecord.OPERATION_METADATA_FIELD) != null);
```

Comment:
Required only for test. Will remove this change.
`hasOperationField` uses the latest data files to know whether operation field is present. The method fails for delete block since delete block is the latest data file but delete block does not store a schema.